File: stack&queue/150. Evaluate Reverse Polish Notation.cpp
#include <iostream>
#include <vector>
#include <stack>

using namespace std;

class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<string> s;

        for(string str: tokens) {
            s.push(str);
            if(str == "+" || str == "-" || str == "*" || str == "/") {

                s.pop();
                int right = stoi(s.top());
                s.pop();
                int left = stoi(s.top());
                s.pop();
                if(str == "+") {
                    s.push(to_string(left + right));
                }  else if(str == "-") {
                    s.push(to_string(left - right));
                }  else if(str == "*") {
                    s.push(to_string(left * right));
                }  else if(str == "/") {
                    s.push(to_string(left / right));
                }
            }
        }

        return stoi(s.top());
    }
};

File: stack&queue/150. Evaluate Reverse Polish Notation.py
from typing import List

class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        
        st = list()
        
        for str in tokens:
            
            if str not in ["+","-","*","/"]:
                st.append(int(str))
            else:
                if str == "+":
                    st.append(st.pop() + st.pop())
                elif str == "-":
                    a = st.pop()
                    b = st.pop()
                    st.append(b - a) 
                elif str == "*":
                    st.append(st.pop() * st.pop())
                elif str == "/":
                    a = st.pop()
                    b = st.pop()
                    st.append((int) (b / a))
                    
        
        return st.pop() 